aptos-move/e2e-move-tests/src/tests/mint_nft.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{assert_success, tests::common, MoveHarness};
+use aptos_crypto::SigningKey;
+use aptos_types::{account_address::AccountAddress, event::EventHandle, state_store::state_key::StateKey};
+use serde::{Deserialize, Serialize};
+use aptos_crypto::ed25519::{Ed25519PublicKey, Ed25519Signature};
+use aptos_types::state_store::table::TableHandle;
+use move_deps::move_core_types::parser::parse_struct_tag;
+
+#[derive(Deserialize, Serialize)]
+struct ModuleData {
+    resource_account_address: AccountAddress,
+}
+
+#[derive(Deserialize, Serialize)]
+struct TokenDataId {
+    creator: AccountAddress,
+    collection: Vec<u8>,
+    name: Vec<u8>,
+}
+
+#[derive(Deserialize, Serialize)]
+struct TokenId {
+    token_data_id: TokenDataId,
+    property_version: u64,
+}
+
+#[derive(Deserialize, Serialize)]
+struct MintProofChallenge {
+    account_address: AccountAddress,
+    module_name: String,
+    struct_name: String,
+    sequence_number: u64,
+    token_data_id: TokenDataId,
+}
+
+#[derive(Deserialize, Serialize)]
+struct TokenStore {
+    tokens: TableHandle,
+    direct_transfer: bool,
+    deposit_events: EventHandle,
+    withdraw_events: EventHandle,
+    burn_events: EventHandle,
+    mutate_token_property_events: EventHandle,
+}
+
+#[test]
+fn mint_nft_e2e() {
+    let mut h = MoveHarness::new();
+    // publish the smart contract @ 0xcafe
+    let acc = h.new_account_at(AccountAddress::from_hex_literal("0xcafe").unwrap());
+    assert_success!(h.publish_package(
+        &acc,
+        &common::test_dir_path("../../../move-examples/mint_nft"),
+    ));
+
+    // getting the module data from the module publisher's (0xcafe) address
+    let resource = h.read_resource::<ModuleData>(acc.address(), parse_struct_tag("0xcafe::minting::ModuleData").unwrap()).unwrap();
```

Comment:
my suggestion was more this: https://github.com/aptos-labs/aptos-core/blob/139cbfc01a58e9b877b3fe2449dfc37e923b1a88/types/src/account_address.rs#L79